examples/camera.rs

```diff
@@ -1,23 +1,113 @@
+use std::collections::HashSet;
+
 use macroquad::*;
 
+fn short_angle_dist(a0: f32, a1: f32) -> f32 {
+    let max = 360.0;
+    let da = (a1 - a0) % max;
+    2.0 * da % max - da
+}
+
+fn angle_lerp(a0: f32, a1: f32, t: f32) -> f32 {
+    a0 + short_angle_dist(a0, a1) * t
+}
+
+fn draw_cross(x: f32, y: f32, color: Color) {
+    let size = 0.1;
+    let thickness= 0.005;
+    draw_line(x - size, y, x + size, y, thickness, color);
+    draw_line(x, y - size, x, y + size, thickness, color);
+}
+
 #[macroquad::main("Camera")]
 async fn main() {
-    loop {
-        clear_background(RED);
+    let mut down_keys: HashSet<KeyCode> = HashSet::new();
 
-        // Render some primitives in camera space
+    let mut target = (0., 0.);
+    let mut zoom = 1.0;
+    let mut rotation = 0.0;
+    let mut smooth_rotation: f32 = 0.0;
+    let mut offset = (0., 0.);
+
+    'main: loop {
+        for &key in get_down_keys(&mut down_keys).iter() {
```

Comment:
I am not quite sold on the idea of `get_down_keys`

As for me, this example would be a bit more readable with just a number of 

```
if is_key_down(Key::Whatever) {
  target.1 -= 0.1;
}
```

Reply:
It was just to write more compact code. Which is easy to add new cases. See diff: https://github.com/not-fl3/macroquad/pull/25/commits/1bd4e3e6364e4aefc8d29a99b0bf9a0ca128bb03#diff-6a6a8253e0291eb499839f6f1f4c15c3L35
However, I removed it.